calico-vpp-agent/cni/cni_server.go

```diff
@@ -506,12 +503,15 @@ func (s *Server) createRedirectToHostRules() (uint32, error) {
 	if err != nil {
 		return types.InvalidID, err
 	}
-	tap0swifindex, tap0nexthop := s.getMainTap0Info()
+	mainInterface := s.getMainInterface()
+	if mainInterface == nil {
+		return types.InvalidID, fmt.Errorf("No main interface found")
+	}
 	for _, rule := range config.GetCalicoVppInitialConfig().RedirectToHostRules {
 		err = s.vpp.AddSessionRedirect(&types.SessionRedirect{
 			FiveTuple:  types.NewDst3Tuple(rule.Proto, net.ParseIP(rule.Ip), rule.Port),
 			TableIndex: index,
-		}, &types.RoutePath{Gw: tap0nexthop, SwIfIndex: tap0swifindex})
+		}, &types.RoutePath{Gw: mainInterface.FakeNextHopIP4, SwIfIndex: mainInterface.TapSwIfIndex})
```

Comment:
I think here we should rather use `config.VppHostPuntFakeGatewayAddress` for session redirection

Reply:
Oh, good catch, thanks !